pkg/distributor/ha_tracker.go

```diff
@@ -462,6 +470,8 @@ func (h *haTracker) updateCache(userID, cluster string, desc *ReplicaDesc) {
 	}
 	if desc.Replica != entry.elected.Replica {
 		h.electedReplicaChanges.WithLabelValues(userID, cluster).Inc()
+		h.lastElectionTimestamp.WithLabelValues(userID, cluster).Set(float64(desc.ReceivedAt / 1000))
+		entry.lastElectionTimestamp = desc.ReceivedAt
```

Comment:
Unfortunately this won't be too precise. Each distributor will show a different value for this, as they will rebuild the cache at different times, which will make the debugging harder.

We should store the lastElectionTimestamp in the KV store to make it consistent.